users/tests/tests.py

```diff
@@ -1,43 +1,60 @@
+import factory
+
 from django.contrib.auth.models import User
 from django.core.urlresolvers import reverse
 from django.test import TestCase, RequestFactory
 
 from users.forms import RegistrationForm
-from users.views import register
+from users.views import UserRegistration
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = User
+
+    username = 'default_username'
+    password = 'pass1234'
 
 
 class RegistrationPageUnitTests(TestCase):
     def setUp(self):
         self.factory = RequestFactory()
 
     def test_method_get(self):
-        request = self.factory.get(reverse('register'))
-        response = register(request)
+        request = self.factory.get(reverse('users:register'))
+        response = UserRegistration.as_view()(request)
         self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.template_name[0], 'registration.html')
 
     def test_method_post(self):
-        request = self.factory.post(reverse('register'))
-        response = register(request)
+        request = self.factory.post(reverse('users:register'))
+        response = UserRegistration.as_view()(request)
         self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.template_name[0], 'registration.html')
```

Comment:
You can remove all the tests since you're testing a unit that comes with Django and is not your unit.
When you use black box units, you don't need to test them. You can rely on integration tests.

The only view unit test you need is for `get_success_url` to make sure the value that's returned is what you expect.